### ISSUE TITLE
Sharepoint rest api -  remove external user 

### ISSUE THREAD

**Issue by CaffeLatte0** [pnp/pnpjs#1185]
### Category
- [x] Enhancement
- [ ] Bug
- [x] Question
- [ ] Documentation gap/issue

Hello,

I'd like to know if it's possible to remove an external user from a folder. 

I granted access to the folder with the **SP.Web.ShareObject** method which accepts the **peoplePickerInput** parameter (whereyou can define all the users to invite).

I found the **SP.Web.UnshareObject** method but it accepts only the item **url** parameter, so you cannot specify the users.

```
pnp.sp.web.unshareObject("https://mysite.sharepoint.com/sites/dev/Docs/test.txt").then((result: SharingResult) => {

    console.log(result);
}).catch(e => {
    console.error(e);
});
```

Is there any solution for this ?

Thanks.

**Comment by koltyakov** [comment]
That sounds correct. Unshare doesn't retract permissions from a specific user but stops document sharing to any previously added shares. It's the projection of the following action in the UI:

![image](https://user-images.githubusercontent.com/7816483/81070900-96b39900-8eec-11ea-9486-b99b64b7ad5c.png)

which has no option to exclude person-wise.

In the UI it's also not possible to exclude a specific individual but to remove the share or share link completely.

![image](https://user-images.githubusercontent.com/7816483/81071220-0cb80000-8eed-11ea-98a3-542f021f0011.png)

Maybe using custom permissions is the case to consider?

However, I'm not deeply familiar with the modern sharing feature. But brief look into the calls and UI I got the impression described above.

**Comment by CaffeLatte0** [comment]
@koltyakov 

Hello Andrew,

Thanks for your reply.

Normally the site owners can do that as you can see from here:

![image](https://user-images.githubusercontent.com/45227282/81074804-d0cf6b80-8ee9-11ea-93d3-35b82b389605.png)

But I couldn't find the related method.

**Comment by koltyakov** [comment]
I got it, thanks, I chose an incorrect share type which doesn't support removing individuals.
Specific people sharing allows such an action. The removing is happening using the same method as for sharing with `inviteesToRemove` property in the payload.
Looks like something which we do not cover at the moment and which can be added as an enhancement:
![image](https://user-images.githubusercontent.com/7816483/81078320-c071bd80-8ef6-11ea-896d-5bfc15be743e.png)
